IdentityCore/src/requests/MSIDSilentTokenRequest.m

```diff
@@ -81,6 +81,7 @@ - (nullable instancetype)initWithRequestParameters:(nonnull MSIDRequestParameter
         _tokenResponseValidator = tokenResponseValidator;
         _tokenResponseHandler = [MSIDTokenResponseHandler new];
         _lastRequestTelemetry = [MSIDLastRequestTelemetry sharedInstance];
+        _throttlingService = [[MSIDThrottlingService alloc] initWithContext:parameters];
```

Comment:
Yes, if we have a sharedInstance I'm afraid of bottleneck. It will also easier to write UT and performance test later (in my bootstrap sso seeding I used sharedInstance and need a lot of mock/ swizzle to able make the test, the test also flakier as it's sharing one instance. )

Reply:
How are requests throttled across entire process in this case?
Agree on unit testing aspect.

Reply:
We got that cover by MSIDLRUCache, as it's a sharedInstance :)